aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/VersionRepository.kt

```diff
@@ -129,32 +130,29 @@ internal class VersionRepository(private val localStorageAdapter: LocalStorageAd
      * @param metadataIterator An iterator of ModelMetadata; the metadata is associated with the provided model
      * @param <T> The type of model
      * @return The version of the model, if available
-     * @throws DataStoreException If there is no version for the model, or if the version cannot be obtained
+     * @throws DataStoreException If there is no metadata for the model
      </T> */
     @Throws(DataStoreException::class)
     private fun <T : Model> extractVersion(
         model: T,
         metadataIterator: Iterator<ModelMetadata>
-    ): Int {
+    ): Optional<Int> {
         val results: MutableList<ModelMetadata> =
-            ArrayList()
+                ArrayList()
         while (metadataIterator.hasNext()) {
             results.add(metadataIterator.next())
         }
-
         // There should be only one metadata for the model....
         if (results.size != 1) {
-            throw DataStoreException(
-                "Wanted 1 metadata for item with id = " + model.primaryKeyString + ", but had " + results.size +
-                    ".",
-                "This is likely a bug. please report to AWS."
+            LOG.warn(
```

Comment:
It sounds like from the description this is not necessarily a bug, right?

Reply:
right, removed the "likely a bug" line. i think keeping the log still makes sense so we have more info in the logs.